### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by MiYaron**
Added OnPush strategy for components and detached components with no changes on the view.
use of switchMap instead of nested subscribe in details component.
Lazy loading details component.

### CODE REVIEW

File: src/app/components/return-button/return-button.component.ts [hunk 1]
```diff
@@ -9,8 +9,13 @@ import { Path } from '../../app.paths';
   templateUrl: './return-button.component.html',
   styleUrl: './return-button.component.css'
 })
-export class ReturnButtonComponent {
+export class ReturnButtonComponent implements OnInit{
   private router = inject(Router);
+  private detectorRef = inject(ChangeDetectorRef);
+
+  public ngOnInit(): void {
+    this.detectorRef.detach();
```
Comment (amiralgazi) [inline src/app/components/return-button/return-button.component.ts:18]: Interesting. 
I've never seen someone who uses this.
Why do you need this?

File: src/app/components/search-bar/search-bar.component.ts [hunk 2]
```diff
@@ -24,7 +25,7 @@ export class SearchBarComponent implements OnInit{
   }
 
   private initFields(): void {
-    this.query = new FormControl('', [Validators.required]);
+    this.query = new FormControl('');
```
Comment (amiralgazi) [inline src/app/components/search-bar/search-bar.component.ts]: Why did you remove the required validation?

File: src/app/pages/home/home.component.html [hunk 3]
```diff
@@ -1,4 +1,8 @@
-<button (click)="addTask()">Add</button>
-<h1>Tusks</h1>
-<app-search-bar></app-search-bar>
-<app-tasks-list></app-tasks-list>
+<main>
+    <div id="header">
+        <div id="logo" [ngStyle]="{ 'height.px': logoHeight }"></div>
```
Comment (amiralgazi) [inline src/app/pages/home/home.component.html]: Please use `rem` instead of `px`

File: src/app/tasks/tasks-list/task-item/task-item.component.html [hunk 4]
```diff
@@ -1,14 +1,15 @@
-<li [id]="task.id">
-  <input
-    [id]="'checkbox' + task.id"
-    type="checkbox"
-    [checked]="task.isDone"
-    (click)="markAsDone()"
-  />
+<li [id]="task.id" [appDrag]="{onDrag: deleteTask.bind(this), onClick: showDetails.bind(this)}">
```
Comment (amiralgazi) [inline src/app/tasks/tasks-list/task-item/task-item.component.html]: Please define the object of appDrag inside the `ts` file and use it in the `html`.
Please also use arrow function instead of `bind` function

File: src/app/pages/home/home.component.ts [hunk 5]
```diff
@@ -7,14 +8,22 @@ import { SearchBarComponent } from '../../components/search-bar/search-bar.compo
 @Component({
   selector: 'app-home',
   standalone: true,
-  imports: [TasksListComponent, SearchBarComponent],
+  imports: [CommonModule, TasksListComponent, SearchBarComponent],
   templateUrl: './home.component.html',
-  styleUrl: './home.component.css'
+  styleUrl: './home.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (amiralgazi) [inline src/app/pages/home/home.component.ts:19]: Please go over all the components in your project and set them as `OnPush`

**Review by amiralgazi — CHANGES_REQUESTED** [review]

File: src/app/components/return-button/return-button.component.ts (hunk 1 above)
Comment (MiYaron) [reply src/app/components/return-button/return-button.component.ts:18]: this is a just a simple button with no changes in the view, thus there is no reason for it to be checked for changes

File: src/app/components/search-bar/search-bar.component.ts (hunk 2 above)
Comment (MiYaron) [reply src/app/components/search-bar/search-bar.component.ts]: it just felt redundant since there is no submit for the query, the filter invokes automatically on input change

**Review by amiralgazi — APPROVED** [review]
